clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3369,6 +3369,13 @@ static void handleIntelNamedSubGroupSize(Sema &S, Decl *D,
                                                                 SizeType)) {
     S.Diag(Loc, diag::warn_attribute_type_not_supported) << AL << SizeStr;
   }
+
+  // If the [[intel::named_sub_group_size]] attribute spelling is used
+  // in earlier SYCL mode (SYCL 2017), we want to diagnose it as being
```

Comment:
What about  - `If the [[intel::named_sub_group_size]] attribute spelling is used in SYCL 2017 mode, we want to.....`